Sources/SwiftProtobuf/Google_Protobuf_Timestamp_Extensions.swift

```diff
@@ -270,49 +266,53 @@ public extension Google_Protobuf_Timestamp {
     public init(decoder: inout JSONDecoder) throws {
         self.init()
         let s = try decoder.scanner.nextQuotedString()
-        let timestamp = try parseTimestamp(s: s)
-        seconds = timestamp.0
-        nanos = timestamp.1
+        (seconds, nanos) = try parseTimestamp(s: s)
     }
 
     public func serializeJSON() throws -> String {
-        let s = seconds
-        let n = nanos
-        if let formatted = formatTimestamp(seconds: s, nanos: n) {
+        if let formatted = formatTimestamp(seconds: seconds, nanos: nanos) {
             return "\"\(formatted)\""
         } else {
             throw EncodingError.timestampJSONRange
         }
     }
 
-    func serializeAnyJSON() throws -> String {
+    public func serializeAnyJSON() throws -> String {
         let value = try serializeJSON()
         return "{\"@type\":\"\(anyTypeURL)\",\"value\":\(value)}"
     }
 }
 
-private func normalizedTimestamp(seconds: Int64, nanos: Int32) -> Google_Protobuf_Timestamp {
-    var s = seconds
-    var n = nanos
-    if n >= 1000000000 || n <= -1000000000 {
-        s += Int64(n) / 1000000000
-        n = n % 1000000000
+private func normalizeForTimestamp(seconds: Int64, nanos: Int32) -> (seconds: Int64, nanos: Int32) {
+    // The Timestamp spec says that nanos must be in the range [0, 999999999), as in actual
+    // modular arithmetic. Integer % and / do not reflect this, so mod() and div() are defined
+    // for this purpose. Factored out, they make normalization very straightforward.
```

Comment:
Why doesn't simple logic like the C++ has work?  https://github.com/google/protobuf/blob/master/src/google/protobuf/util/time_util.cc#L62-L72